### PR TITLE
[IMP] l10n_ar_edi_ux: agregar funcionalidad de debitar cheques propios.

### PR DESCRIPTION
**Pull request by pablohmontenegro**
Tarea: 34995
Lo hacemos acá y no en l10n_ar_ux porque no tenemos enterprise y usamos acá un metodo de account_accountant.

### CODE REVIEW

**Comment by roboadhoc** [comment]
[Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/odoo-argentina-ee/pull/297).

File: l10n_ar_edi_ux/wizards/account_check_action_wizard.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        required=True,
+    )
+
+    def action_confirm(self):
+        """ Este método sirve para hacer el débito de cheques con cuenta outstanding desde los payments con método de pago de cheques. """
+        payment = self.env['account.payment'].browse(self._context.get('active_id', False))
+        if self.date < payment.date:
+            raise ValidationError(f'La fecha del débito del cheque {self.date} no puede ser inferior a la fecha de emisión del mismo {payment.date}.')
+        # Línea del cheque a conciliar.
+        move_line_ids = [payment.line_ids.filtered(lambda x: not x.reconciled and x.account_type=='liability_current').id]
```
Comment (jjscarafia) [inline l10n_ar_edi_ux/wizards/account_check_action_wizard.py]: usaría el seek for lines porque puede tener otro tipo

File: l10n_ar_edi_ux/wizards/account_check_action_wizard.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        new_mv_line_dicts = [{'name': f'Débito cheque nro {payment.check_number}',
+                            'balance': abs(payment.line_ids.filtered(lambda x: not x.reconciled and x.account_type=='liability_current').balance),
+                            'account_id': outstanding_account.id,
+                            'journal_id': payment.journal_id.check_debit_journal_id.id,
+                            'date': self.date,
+                            }]
+        self.env['account.reconciliation.widget']._process_move_lines(move_line_ids, new_mv_line_dicts)
+        payment.message_post(body=f'El cheque nro "{payment.check_number}" ha sido debitado.')
+
+    def _get_outstanding_account(self, payment):
```
Comment (jjscarafia) [inline l10n_ar_edi_ux/wizards/account_check_action_wizard.py:38]: usaría algo más parecido a esto https://github.com/odoo/odoo/blob/16.0/addons/account/models/account_payment.py#L490

File: l10n_ar_edi_ux/__manifest__.py [hunk 3]
```diff
@@ -17,11 +17,13 @@
     'data': [
         'wizards/res_config_settings_view.xml',
         'wizards/res_partner_update_from_padron_wizard_view.xml',
-        'wizards/account_payment_group_invoice_wizard_view.xml',
+        'wizards/account_check_action_wizard_view.xml',
```
Comment (jjscarafia) [inline l10n_ar_edi_ux/__manifest__.py:21]: agregar dependencia a account_accountant

File: l10n_ar_edi_ux/models/account_journal.py [hunk 4]
```diff
@@ -62,3 +67,10 @@ def _format_afip_doc_types(self, ws, response):
                 line += " hasta: " + date_to
             msg += line + "\n"
         return msg
+
+    @api.onchange('check_add_debit_button')
```
Comment (jjscarafia) [inline l10n_ar_edi_ux/models/account_journal.py]: check_debit_journal_id = fields.Many2one('account.journal', compute='_compute_check_debit_journal', store=True, readonly=False, domain="[('company_id', '=', company_id), ('type', '=', 'general')]", help="Debe seleccionar un diario de tipo 'Varios' donde se realizará el asiento del débito")

    @api.depends('check_add_debit_button')
    def _compute_check_debit_journal(self):
        """ En caso de que el campo 'Agregar botón de débito' (check_debit_journal_id) sea 'False' entonces debemos asegurarnos que el campo check_debit_journal_id esté vacío para evitar inconvenientes en caso de necesitar debitar un cheque. """
        for journal in self.filtered(lambda x: not x.check_add_debit_button):
            journal.check_debit_journal_id = False

File: l10n_ar_edi_ux/wizards/account_check_action_wizard.py (hunk 1 above)
Comment (jjscarafia) [reply l10n_ar_edi_ux/wizards/account_check_action_wizard.py]: _seek_for_lines (utilizando

liquidity_lines, counterpart_lines, writeoff_lines = payment._seek_for_lines()
move_line_ids = liquidity_lines.ids

File: l10n_ar_edi_ux/wizards/account_check_action_wizard.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        payment = self.env['account.payment'].browse(self._context.get('active_id', False))
+        if self.date < payment.date:
+            raise ValidationError(f'La fecha del débito del cheque {self.date} no puede ser inferior a la fecha de emisión del mismo {payment.date}.')
+        # Línea del cheque a conciliar.
+        move_line_ids = [payment.line_ids.filtered(lambda x: not x.reconciled and x.account_type=='liability_current').id]
+        # Obtenemos la cuenta outstanding del método de pago pentientes "manual" del diario del pago o bien la "Cuenta de pagos pentientes" de la compañía.
+        outstanding_account = self._get_outstanding_account(payment)
+        # Obtenemos fecha, importe, pasamos cuenta outstanding y el diario es de tipo varios que está vinculado al diario del pago del cheque (ver campo Check Debit Journal en el diario) para que nos permita hacer la conciliación. Lo utilizamos como un diario "puente".
+        new_mv_line_dicts = [{'name': f'Débito cheque nro {payment.check_number}',
+                            'balance': abs(payment.line_ids.filtered(lambda x: not x.reconciled and x.account_type=='liability_current').balance),
```
Comment (jjscarafia) [inline l10n_ar_edi_ux/wizards/account_check_action_wizard.py]: abs(sum(liquidity_lines.mapped('balance'))

File: l10n_ar_edi_ux/wizards/account_check_action_wizard.py (hunk 2 above)
Comment (jjscarafia) [reply l10n_ar_edi_ux/wizards/account_check_action_wizard.py:38]: journal = payment.journal_id
journal_manual_payment_method = journal.outbound_payment_method_line_ids.filtered(lambda x: x.code=='manual')
outstanding_account = (journal_manual_payment_method.payment_account_id or journal.company_id.account_journal_payment_credit_account_id)
if not outstanding_account:
    raise UserError(_(
        "You can't create a new payment without an outstanding payments/receipts account set either on the company or the %s payment method in the %s journal.",
        self.payment_method_line_id.name, self.journal_id.display_name))

**Review by jjscarafia — CHANGES_REQUESTED** [review]

**Comment by jjscarafia** [comment]
@roboadhoc r+

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @jjscarafia this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina-ee#300

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @jjscarafia this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina-ee#300

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @jjscarafia this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina-ee#300

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @jjscarafia this pull request has forward-port PRs awaiting action (not merged or closed):
- ingadhoc/odoo-argentina-ee#300

**Pull request closed without merging**